integration/docker/entrypoint.sh

```diff
@@ -113,41 +94,126 @@ function mountAlluxioRootFSWithFuseOption {
   tail -f /opt/alluxio/logs/fuse.log
 }
 
-case ${service,,} in
-  master)
-    formatMasterIfSpecified
-    integration/docker/bin/alluxio-job-master.sh &
-    integration/docker/bin/alluxio-master.sh &
-    wait -n
-    ;;
-  master-only)
-    formatMasterIfSpecified
-    integration/docker/bin/alluxio-master.sh
-    ;;
-  job-master)
-    integration/docker/bin/alluxio-job-master.sh
-    ;;
-  worker)
-    formatWorkerIfSpecified
-    integration/docker/bin/alluxio-job-worker.sh &
-    integration/docker/bin/alluxio-worker.sh &
-    wait -n
-    ;;
-  worker-only)
-    formatWorkerIfSpecified
-    integration/docker/bin/alluxio-worker.sh
-    ;;
-  job-worker)
-    integration/docker/bin/alluxio-job-worker.sh
-    ;;
-  proxy)
-    integration/docker/bin/alluxio-proxy.sh
-    ;;
-  fuse)
-    mountAlluxioRootFSWithFuseOption
-    ;;
-  *)
+# Sends a signal to each of the running background processes
+#
+# Args:
+#     1: the signal to send
+function forward_signal {
+  local signal
+  signal="${1}"
+
+  # background jobs don't respond to SIGINT
+  # Change to SIGHUP
+  if [ "${signal}" -eq "2" ]; then
+    signal="1"
+  fi
+
+  local procs="$(jobs -p)"
```

Comment:
how does this work? seems like you launch the background processes after setting up signals

Reply:
Here's the general flow I had in my head.

```
- Must launch either 1 process or 2 processes
  - if 1, then simply exec and the signal handlers don't need to be set up. All signals will go straight to the process
- If 2 processes, then a signal must be forwarded to both background processes
  - To forward a signal to 2 processes, set up all the traps within the script first
  - Then launch the background processes
  - Any signals after this point get forwarded
```

Reply:
Also,

calling `trap` is like setting up a signal handler. Basically the function doesn't get called _until_ the signal is sent.

so calling `trap "forward_signal 1" 1` only executes the `forward_signal 1` command when `1` is sent to the shell process.

Reply:
got it, thx